src/Components/allocations/helper_functions/deleteAllocation.ts

```diff
@@ -0,0 +1,10 @@
+import { Dispatch, SetStateAction } from "react";
+import { updateDataInFirebase } from "../../../dataBaseUtils/readWrite";
+import { allocationType } from "../../../types";
+
+export default function deleteAllocation(allocation: allocationType, setReRender: Dispatch<SetStateAction<number>>): void {
+  updateDataInFirebase(`allocations/${allocation.id}`, null)
+    .then(() => {
+      setReRender(prev => prev + 1);
```

Comment:
This is a violation of the sharing responsibility principles. Function `deleteAllocation` all of a sudden starts to know about things it doesn't really want to know at all (`setReRender`). Just make it `async` and handle promise in the parent, awaiting for it and calling `setReRender` there.